cmd/serve/sftp/sftp.go

```diff
@@ -101,6 +112,22 @@ sftp backend, but it may not be with other SFTP clients.
 		}
 		cmd.Run(false, true, command, func() error {
 			s := newServer(context.Background(), f, &Opt)
+			if Opt.Stdio {
+				if terminal.IsTerminal(int(os.Stdout.Fd())) {
+					return errors.New("Refusing to run SFTP server directly on a terminal. Please let sshd start rclone, by connecting with sftp or sshfs.")
+				}
+				sshChannel := &StdioChannel{
+					stdin: os.Stdin,
+					stdout: os.Stdout,
+				}
+				handlers := newVFSHandler(vfs.New(f, &vfsflags.Opt))
+				server := sftp.NewRequestServer(sshChannel, handlers)
+				defer server.Close()
+				if err := server.Serve(); err != nil {
+					return errors.Wrap(err, "failed to start stdio server")
+				}
```

Comment:
Is it worth deduplicating this and https://github.com/rclone/rclone/blob/71f75a1d95a0154a6d77e725e097ed733a1e2db9/cmd/serve/sftp/connection.go#L229-L241 ?

Reply:
Factoring that out of `handleChannel` seems like a good idea since `handleChannel` is rather long anyway! Maybe into a `serveChannel` function? It should return a non `nil` error if it gets to the `fs.Errorf` case.

I'd include the `fs.Debugf(c.what, "Starting SFTP server")` line too.